documentation/src/docs/asciidoc/client/task-chain.adoc

```diff
@@ -44,40 +44,63 @@ backgroundExecutors.execute(() -> {
-The TaskChain provides a fluent API that let you define workflows with multiple switches between background threads and the ui thread.
+The TaskChain provides a fluent API that let you define workflows with multiple switches between background threads and the UI thread.
 Before we have a deeper look at the API let's see how the given example would look like by using the TaskChain API:
 
 [source,java]
 ----
-UiExecutor.createUiTaskChain()
-    .execute(() -> showLoadingAnimation())
-    .background()
-    .supply(() -> callServer())
-    .ui()
-    .consume(v -> updateUi(v))
-    .onException(e -> showErrorDialog(e))
-    .thenFinally(() -> hideLoadingAnimation())
-    .run();
+UiExecutor.createUiTaskChain() // <1>
+    .execute(() -> showLoadingAnimation()) <2>
+    .background() // <3>
+    .supply(() -> callServer()) <4>
+    .map(v -> v.getName()) // <5>
+    .ui() // <6>
+    .consume(v -> updateUi(v)) // <7>
+    .onException(e -> showErrorDialog(e)) // <8>
+    .thenFinally(() -> hideLoadingAnimation()) // <9>
+    .run(); // <10>
 ----
 
-////
-todo: some description of the code
-////
+<1> a UiTaskChain is will execute the tasks in the UI thread. There is also a BackgroundTaskChain.
+<2> the `execute()` method adds a runnable to the chain.
+<3> by calling `background()` the execution for the coming tasks is switched to a background thread.
+<4> the `supply()` method adds a supplier to the chain. The result of the supplier is available to the next task.
+<5> the `map()` method adds a function to the chain. The result of the function is available to the next task.
+<6> by calling `ui()` the execution for the coming tasks is switched to the UI thread.
+<7> the `consume()` method adds a consumer to the chain.
+<8> the `onException()` method adds an exception handler to the chain.
+<9> the `thenFinally()`method adds a runnable to the chain. The finally task is called in any case.
+<10> the chain is only executed when `run()` is called. This allows to create a chain and run it later or even run it multiple times.
+
+The `run()` method returns a CompletableFuture which allows the caller of `run()` to wait for the completion of all tasks in the chain.
+Also the caller can determine if the tasks chain completed successfully or not by inspecting the CompletableFuture.
+Finally if the output from a task is not consumed by a consumer in the chain than this value can be obtained from the CompletableFuture.
 
 The main entry point of the TaskChain API is the `dev.rico.client.concurrent.TaskChain` interface.
 New instances should always be created by factory methods that are provided by the `dev.rico.client.concurrent.UiExecutor` interface.
 
 [source,java]
 ----
-final TaskChain chain = UiExecutor.createUiTaskChain();
+final TaskChain uiChain = UiExecutor.createUiTaskChain();
+final TaskChain bgChain = UiExecutor.createBackgroundTaskChain();
 ----
 
-////
+==== Exception handling in the task chain
```

Comment:
in all other docs we have an empty line after a header. makes it better readable for me ;) What do you think?